gnssanalysis/gn_io/sp3.py

```diff
@@ -141,6 +147,18 @@ def read_sp3(sp3_path, pOnly=True):
 
     sp3_df.attrs["HEADER"] = parsed_header  # writing header data to dataframe attributes
     sp3_df.attrs["path"] = sp3_path
+
+    # Check for duplicate epochs, dedupe and log warning
+    duplicated_indexes = sp3_df.index.duplicated()
+    if duplicated_indexes.sum() > 0: # We have dupes
+        first_dupe = sp3_df.index.get_level_values(0)[duplicated_indexes][0]
```

Comment:
Not sure if it is worth it to list out all indexes or rows that are copied out (to show the scale of the problem?)
Perhaps doing something like `sp3_df.index.get_level_values(0)[duplicated_indexes].sum() / 2` should reveal the number of entries that are copied? Not entirely sure on this so it would need to be tested

Reply:
Updated to output the count in the warning. By the doc, it's by default the number of entries which duplicate something already there. So a (possibly repeated) list of duplicated values. I've worded the warning to try and capture that.